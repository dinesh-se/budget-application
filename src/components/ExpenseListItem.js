import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import { startRemoveExpense } from '../actions/expenses';

export const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
  <div>
    <h3>
      <Link to={`/edit/${id}`}>{description}</Link>
    </h3>
    <p>
      {numeral(amount / 100).format('$0,0.00')}
      -
      {moment(createdAt).format('MMMM Do, YYYY')}
    </p>
    <button onClick={() => {
      dispatch(startRemoveExpense(id));
    }}>
      Remove</button>
  </div>
);

export default connect()(ExpenseListItem);