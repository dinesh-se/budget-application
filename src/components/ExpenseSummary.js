import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getTotalExpenses from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = (props) => {
  return (
    <div>
      <h5>Total expense: {numeral(getTotalExpenses(props.expenses) / 100).format('$0,0.00')}</h5>
    </div>
  )
};

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters) 
});

export default connect(mapStateToProps)(ExpenseSummary);