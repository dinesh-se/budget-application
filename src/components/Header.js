import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Family Budget</h1>
    <ul>
      <li>
        <NavLink to='/' activeClassName='is-active' exact={true}>Home</NavLink>
      </li>
      <li>
        <NavLink activeClassName='is-active' to='/create'>Add Expense</NavLink>
      </li>
      <li>
        <NavLink activeClassName='is-active' to='/help'>Help</NavLink>
      </li>
    </ul>
  </header>
);

export default Header;