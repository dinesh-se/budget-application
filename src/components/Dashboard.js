import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';

const BudgetDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
    <ExpenseSummary />
  </div>
);

export default BudgetDashboardPage;