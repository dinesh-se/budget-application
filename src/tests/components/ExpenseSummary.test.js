import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should correctly render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpensesSummary with multiple expense', () => {
  const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});