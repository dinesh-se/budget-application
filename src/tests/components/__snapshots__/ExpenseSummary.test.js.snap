// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h5>
    Total expense: 
    $15,000.00
  </h5>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expense 1`] = `
<div>
  <h5>
    Total expense: 
    $116,500.00
  </h5>
</div>
`;
